### DISCUSSION TITLE
How Hosting React next App with Firebase works?

### DISCUSSION THREAD

**Discussion by Leonardll** [vercel/next.js#31438] [Help]
### What version of Next.js are you using?

next": "^12.0.4-canary.4"

### What version of Node.js are you using?

14.16.0

### What browser are you using?

chrome

### What operating system are you using?

windows

### How are you deploying your application?

npm run build  "next build && next export

### Describe the Bug

i am trying to deploy my react next app with firebase but i am running into a number of issues. When i deploy run firebase deploy after running firebase init and npm run build. i have tried also to add the && export to json script but nothing works. It doesn't generate a static html page with my contents in.

i am getting the below error message. When i run npm run start my page render on localhost with no issue but it does not resolve deployment issue. I am still not getting my static html.

λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)

info  - using build directory: C:\Users\lonar\Desktop\Project\crypto-react\.next
info  - Copying "static build" directory
info  - No "exportPathMap" found in "C:\Users\lonar\Desktop\Project\crypto-react\next.config.js". Generating map from "./pages"
Error: Image Optimization using Next.js' default loader is not compatible with `next export`.
  Possible solutions:
    - Use `next start` to run a server, which includes the Image Optimization API.
    - Use any provider which supports Image Optimization (like Vercel).
    - Configure a third-party loader in `next.config.js`.
    - Use the `loader` prop for `next/image`.
  Read more: https://nextjs.org/docs/messages/export-image-api
    at C:\Users\lonar\Desktop\Project\crypto-react\node_modules\next\dist\export\index.js:256:23
    at async Span.traceAsyncFn (C:\Users\lonar\Desktop\Project\crypto-react\node_modules\next\dist\trace\trace.js:74:20)

I have put below my current next.config.js . Can someone help me set it up for it to run,please?

const webpack = require('webpack');
const path = require('path');
//const withPlugins = require('next-compose-plugins');
//const optimizedImages = require('next-optimized-images');

module.exports = {
  images: {
    domains: ['assets.coingecko.com'],
    //loader: 'imgix',
    //path: 'https://assets.coingecko.com/',
  },
  reactStrictMode: true,
  entry: './src/index.js',
  sassOptions: {
    includePaths: [path.join(__dirname, 'styles')],
  },
  module: {
    rules: [
      //...
      {
        test: /\.(png|jp(e*)g|svg|gif)$/,
        use: [
          {
            loader: 'file-loader',
            options: {
              name: 'images/[hash]-[name].[ext]',
            },
          },
        ],
      },
    ],
  },

  //...

### Expected Behavior

I would expext to have static html generated so i can deploy my app on firebase 

### To Reproduce

https://github.com/Leonardll/crypto-react

**Comment by timneutkens** [comment]
The error links to https://nextjs.org/docs/messages/export-image-api which explains what to do
```
Error: Image Optimization using Next.js' default loader is not compatible with next export.
Possible solutions:
- Use next start to run a server, which includes the Image Optimization API.
- Use any provider which supports Image Optimization (like Vercel).
- Configure a third-party loader in next.config.js.
- Use the loader prop for next/image.
Read more: nextjs.org/docs/messages/export-image-api
```

**Reply by Leonardll** [reply]
Thanks for helping. i have went through this documention but it does not address the matter clearly. I am still struggling to fetch the images i beleive the issue comes from the  setting of images in next.config. It is not clear how to set the url path in the loader for it to render. I have asked a new question related to this topic maybe you could help?
